src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyAnalyzer.cs

```diff
@@ -0,0 +1,117 @@
+        {
+            context.RegisterSyntaxNodeAction(snac => AnalyzeArgument(ineligibleFields, snac), SyntaxKind.Argument);
+        }
+
+        private void AnalyzeArgument(ConcurrentBag<IFieldSymbol> ineligibleFields, SyntaxNodeAnalysisContext context)
+        {
+            // An argument will disqualify a field if that field is used in a ref/out position.  
+            // We can't change such field references to be property references in C#.
+            var argument = (ArgumentSyntax)context.Node;
+            if (argument.RefOrOutKeyword.Kind() == SyntaxKind.None)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(argument.Expression, cancellationToken);
+            AddIneligibleField(symbolInfo.Symbol, ineligibleFields);
+            foreach (var symbol in symbolInfo.CandidateSymbols)
+            {
+                AddIneligibleField(symbol, ineligibleFields);
+            }
+        }
+
+        private static void AddIneligibleField(ISymbol symbol, ConcurrentBag<IFieldSymbol> ineligibleFields)
+        {
+            var field = symbol as IFieldSymbol;
+            if (field != null)
+            {
+                ineligibleFields.Add(field);
+            }
+        }
+
+        private bool CheckExpressionSyntactically(ExpressionSyntax expression)
+        {
+            if (expression?.Kind() == SyntaxKind.SimpleMemberAccessExpression)
+            {
+                var memberAccessExpression = (MemberAccessExpressionSyntax)expression;
+                return memberAccessExpression.Expression.Kind() == SyntaxKind.ThisExpression &&
+                    memberAccessExpression.Name.Kind() == SyntaxKind.IdentifierName;
+            }
+            else if (expression.Kind() == SyntaxKind.IdentifierName)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        protected override ExpressionSyntax GetGetterExpression(IMethodSymbol getMethod, CancellationToken cancellationToken)
+        {
+            var getAccessor = getMethod.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken) as AccessorDeclarationSyntax;
+            var firstStatement = getAccessor?.Body.Statements.SingleOrDefault();
+            if (firstStatement?.Kind() == SyntaxKind.ReturnStatement)
+            {
+                var expr = ((ReturnStatementSyntax)firstStatement).Expression;
+                return CheckExpressionSyntactically(expr) ? expr : null;
+            }
+
+            return null;
+        }
+
+        protected override ExpressionSyntax GetSetterExpression(IMethodSymbol setMethod, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            var setAccessor = setMethod.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken) as AccessorDeclarationSyntax;
+
+            // Setter has to be of the form:
```

Comment:
What about static properties and fields? In that case, we'd also care about the left side because of the form, ```set { Namespace.Class.field = value; }```